torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4288,6 +4307,73 @@ def sample_inputs_narrow(op_info, device, dtype, requires_grad, **kwargs):
         yield sample
         yield SampleInput(sample.input, args=(sample.args[0], torch.tensor(sample.args[1]), sample.args[2]))
 
+def error_inputs_narrow_narrow_copy(op_info, device):
+    make_arg = partial(make_tensor, device=device, dtype=torch.float32)
+
+    # shape, args, error_type, error_regex
+    samples = [
+        # 0-dim
+        ((), (0, 0, 1), RuntimeError,
+         r"narrow\(\) cannot be applied to a 0-dim tensor\."),
+
+        # out of bounds dim is tested below
+        # out of bounds dim (negative)
+        ((L, S, M), (-4, 0, 0), IndexError,
+         r"Dimension out of range \(expected to be in range of \[-3, 2\], but got -4\)"),
+
+        # out of bounds start is tested below
+        # out of bounds start (negative), negative start means counting from the right
+        ((L, M, S), (1, -M - 1, 0), IndexError,
+         r"Dimension out of range \(expected to be in range of \[-10, 9\], but got -11\)"),
+
+        # out of bounds length
+        ((S, L, M), (2, 0, M + 1), RuntimeError,
+         r"start \(0\) \+ length \(11\) exceeds dimension size \(10\)\."),
+        # out of bounds length (negative)
+        ((M,), (0, 0, -1), RuntimeError,
+         r"start \(0\) \+ length \(-1\) exceeds dimension size \(10\)\."),
+    ]
+
+    # narrow_copy_dense_cpu_out in core has slightly different checks:
+    # https://github.com/pytorch/pytorch/issues/87019
+    # out of bounds dim
+    if op_info.name == 'narrow_copy' and device == 'cpu':
```

Comment:
Ideally we wouldn't check the operation name, although I understand why it's convenient here. Might be worth just addressing the consistency issue?

To check for the device type, use `torch.device(device).type == 'cpu'` instead of assuming the device is a string. That particular check will work fine today on CPU, but not on other device types or if we change how `device` is represented in the test suite